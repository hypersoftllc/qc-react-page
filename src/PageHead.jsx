import PT from 'prop-types'
import React from 'react'
import Block from '@qc/react-block'


function PageHead(props) {
  const domProps = {
    ...props,
    className: `${props.compClassName} ${props.className}`.trim(),
  }
  delete domProps.children
  delete domProps.compClassName
  return (
    <Block {...domProps}>
      {props.children}
    </Block>
  )
}

PageHead.defaultProps = {
  className: '',
  compClassName: 'Page-Head',
  compType: 'header',
}

PageHead.propTypes = {
  children: PT.node,
  className: PT.string,
  compClassName: PT.string,
  compType: PT.oneOf(['div', 'header', 'section']),
}

export default PageHead
