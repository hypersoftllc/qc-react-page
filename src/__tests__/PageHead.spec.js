/* eslint-env jest */
/* eslint padded-blocks: "off" */

import React from 'react'
import TestRenderer from 'react-test-renderer'

import PageHead from '../PageHead'


describe('PageHead', () => {

  it('should be a function', () => {
    expect(typeof PageHead).toBe('function')
  })

  it('should be a functional component', () => {
    let comp

    comp = TestRenderer.create(
      <PageHead/>
    )

    // Functional components don't have instances.
    expect(comp.getInstance()).toBe(null)
  })

  describe('rendered', () => {

    describe('with no props', () => {

      it('should use default props', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageHead/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with children', () => {

      it('should render children', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageHead>
            <span>Child</span>
          </PageHead>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with className', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageHead className="Foo"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compClassName', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageHead compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageHead className="foo" compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageHead className="foo" compClassName="page-head"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compType', () => {

      it('should render with expected component', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageHead compType="div"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageHead compType="header"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageHead compType="section"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

  })

})
