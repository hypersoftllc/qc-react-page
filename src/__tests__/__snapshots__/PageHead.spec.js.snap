// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHead rendered with children should render children 1`] = `
<header
  className="Block Page-Head"
>
  <span>
    Child
  </span>
</header>
`;

exports[`PageHead rendered with className should render with expected class attribute 1`] = `
<header
  className="Block Page-Head Foo"
/>
`;

exports[`PageHead rendered with compClassName should render with expected class attribute 1`] = `
<header
  className="Block"
/>
`;

exports[`PageHead rendered with compClassName should render with expected class attribute 2`] = `
<header
  className="Block  foo"
/>
`;

exports[`PageHead rendered with compClassName should render with expected class attribute 3`] = `
<header
  className="Block page-head foo"
/>
`;

exports[`PageHead rendered with compType should render with expected component 1`] = `
<div
  className="Block Page-Head"
/>
`;

exports[`PageHead rendered with compType should render with expected component 2`] = `
<header
  className="Block Page-Head"
/>
`;

exports[`PageHead rendered with compType should render with expected component 3`] = `
<section
  className="Block Page-Head"
/>
`;

exports[`PageHead rendered with no props should use default props 1`] = `
<header
  className="Block Page-Head"
/>
`;
